src/learning/WeightUpdater.cpp

```diff
@@ -0,0 +1,112 @@
+#include "../../include/learning/WeightUpdater.h"
+#include "../../include/learning/LinearSolver.h"
+
+#include <sstream>
+
+WeightUpdater::WeightUpdater(AlternativesPerformance &ap, Config &conf)
+    : ap(ap), conf(conf), solver(ap, conf) {}
+
+WeightUpdater::WeightUpdater(const WeightUpdater &wu)
+    : solver(wu.solver), ap(wu.ap), conf(wu.conf) {}
+
+WeightUpdater::~WeightUpdater() {}
+
```

Comment:
Yes, the LinearSolver class is stored by value, thus it will be deleted when the WeightUpdater is deleted. The ortools linear solver being stored by pointer has to be deleted manually, that's why we have it in the LinearSolver destructor